addons/crm/views/crm_lead_views.xml

```diff
@@ -412,35 +412,99 @@
             <field name="arch" type="xml">
                 <form>
                     <group>
-                        <field name="partner_id" widget="res_partner_many2one"
-                            class="o_field_highlight"
+                    <div class="d-flex align-items-center">
+                        <i class="fa fa-building me-2" title="Company"/>
+                        <field name="company_partner_id"
+                            placeholder="Company"
+                            string="Company"
+                            nolabel="1"
+                            class="o_field_highlight w-100 mb-0"
+                            context="{
+                            'default_is_company': True,
+                            'res_partner_search_mode': type == 'opportunity' and 'customer' or False,
+                            'default_name': contact_name or partner_name,
+                            'default_is_company': True,
+                            'default_type': 'contact',
+                            'default_phone': phone,
+                            'default_email': email_from,
+                            'default_user_id': user_id,
+                            'default_team_id': team_id,
```

Comment:
If we create a company, and use the contact_name flow, both the company and the partner_id will have their email and phone set to the same values, right?